### PR TITLE
UCT/MODULES: added .pc files

### PR DESCRIPTION
**Pull request by hoopoepg**
- added module-specific .pc files to build

### CODE REVIEW

**Review by yosefe — COMMENTED** [review]
a lot of duplication - can we make it templated?

**Comment by hoopoepg** [comment]
> a lot of duplication - can we make it templated?

unfortunately no without any hacked ways (like copy/create FS link in autogen.sh)
removed duplicated sections from .pc.in files

**Comment by yosefe** [comment]
> > a lot of duplication - can we make it templated?
> 
> unfortunately no without any hacked ways (like copy/create FS link in autogen.sh) removed duplicated sections from .pc.in files

how about specifying in files manually?
https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Configuration-Files.html

**Comment by hoopoepg** [comment]
problems with customisation: I don't see way to create different contents from same template.
.pc.in file may include variables which are substituted from .m4 files, but such substitution is global

**Comment by shamisp** [comment]
seems like ugni and xpmem are not there

**Comment by hoopoepg** [comment]
> seems like ugni and xpmem are not there

right, added only base modules. Cuda/rocm/xpmem/ugni will be added later

File: src/ucs/vfs/fuse/Makefile.am [hunk 1]
```diff
@@ -14,6 +14,11 @@ libucs_fuse_la_LIBADD   = $(FUSE3_LIBS) \
                           $(top_builddir)/src/ucs/libucs.la
 libucs_fuse_la_LDFLAGS  = $(FUSE3_LDFLAGS) -version-info $(SOVERSION)
 libucs_fuse_la_SOURCES  = vfs_fuse.c
+
+EXTRA_DIST = ucx-fuse.pc.in
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = ucx-fuse.pc
```
Comment (yosefe) [inline src/ucs/vfs/fuse/Makefile.am:20]: as discussed offline - pls try to move these inside `$(top_srcdir)/config/module.am`

File: src/ucs/vfs/fuse/ucx-fuse.pc.in [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Name: @PACKAGE@-fuse
+Description:  Unified Communication X Library FUSE module
```
Comment (yosefe) [inline src/ucs/vfs/fuse/ucx-fuse.pc.in:2]: extra space before Unified

File: src/uct/ib/rdmacm/ucx-rdmacm.pc.in [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Name: @PACKAGE@-rdmacm
+Description:  Unified Communication X Library RDMACM module
```
Comment (yosefe) [inline src/uct/ib/rdmacm/ucx-rdmacm.pc.in:2]: extra space before Unified (in few places)

**Comment by shamisp** [comment]
> > seems like ugni and xpmem are not there
> 
> right, added only base modules. Cuda/rocm/xpmem/ugni will be added later

Why not to add everything now ?

File: src/ucs/vfs/fuse/Makefile.am (hunk 1 above)
Comment (hoopoepg) [reply src/ucs/vfs/fuse/Makefile.am]: there are issues with conditional processing of makefiles, added separate include file to identify pkg-config variables

File: src/ucs/vfs/fuse/ucx-fuse.pc.in (hunk 2 above)
Comment (hoopoepg) [reply src/ucs/vfs/fuse/ucx-fuse.pc.in]: removed extra space

File: src/uct/ib/rdmacm/ucx-rdmacm.pc.in (hunk 3 above)
Comment (hoopoepg) [reply src/uct/ib/rdmacm/ucx-rdmacm.pc.in]: removed extra space

**Comment by hoopoepg** [comment]
> Why not to add everything now ?

some modules need special environment to build. used modules which can be built on any host in lab to simplify investigations

**Review by yosefe — APPROVED** [review]

**Comment by yosefe** [comment]
@hoopoepg can you pls add missing pkgconfig files in next PR?

**Comment by hoopoepg** [comment]
done: https://github.com/openucx/ucx/pull/7802

**Pull request merged**